### ISSUE TITLE
Last update to 2.5.107 made csshell disfunctional.

### ISSUE THREAD

**Issue by Alisis33** [radimitrov/CSharpShellApp#163]
The last update to 2.5.107 made csshell disfunctional.
Down the error message.

I would like mention a idea I described in a other post.
The fore icons on the top bar could open each a pulldown with the funktions
Copy/past/...
Undo/redo/search/..
Views/...
Copiles/...

The icons could be big and so not missclicked, they can hafe text in the pulldown, and all would be allways at the same place and all operations only 2 clicks away, no long-clicks, easy to go in to muscle-memory.
If this could be a option, I would probable prefer this.

After selecting or coping text:

Java.Lang.IllegalStateException: Observer androidx.recyclerview.widget.RecyclerView$RecyclerViewDataObserver@37d8f1 was not registered.
  at Java.Interop.JniEnvironment+InstanceMethods.CallNonvirtualVoidMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x0008e] in <1921523bc22e407fa9a0855bdae29335>:0 
  at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeVirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x00063] in <1921523bc22e407fa9a0855bdae29335>:0 
  at AndroidX.RecyclerView.Widget.RecyclerView.SetAdapter (AndroidX.RecyclerView.Widget.RecyclerView+Adapter adapter) [0x00031] in <3cd33b68cea545ab982e8a156929cccd>:0 
  at CSharp_Shell.Activities.MainActivity.UpdateQuickbar () [0x0031b] in <52e3c3563650487db862f4946bef1275>:0 
  at CSharp_Shell.Activities.MainActivity.HideCopyPaste () [0x00047] in <52e3c3563650487db862f4946bef1275>:0 
  at CSharp_Shell.Activities.MainActivity.<UpdateQuickbar>b__199_4 () [0x00010] in <52e3c3563650487db862f4946bef1275>:0 
  at CSharp_Shell.IDE.Views.ActionRecyclerAdapter.Text_Click (System.Object sender, System.EventArgs e) [0x00020] in <52e3c3563650487db862f4946bef1275>:0 
  at Android.Views.View+IOnClickListenerImplementor.OnClick (Android.Views.View v) [0x00011] in <d6d61ecb611a47bdb0740fd[298231936](tel:298231936)>:0 
  at Android.Views.View+IOnClickListenerInvoker.n_OnClick_Landroid_view_View_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_v) [0x0000f] in <d6d61ecb611a47bdb0740fd[298231936](tel:298231936)>:0 
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.75(intptr,intptr,intptr)
  --- End of managed Java.Lang.IllegalStateException stack trace ---
java.lang.IllegalStateException: Observer androidx.recyclerview.widget.RecyclerView$RecyclerViewDataObserver@37d8f1 was not registered.
	at android.database.Observable.unregisterObserver([Observable.java:69](http://observable.java:69/))
	at androidx.recyclerview.widget.RecyclerView$Adapter.unregisterAdapterDataObserver([RecyclerView.java:7531](http://recyclerview.java:7531/))
	at androidx.recyclerview.widget.RecyclerView.setAdapterInternal([RecyclerView.java:1232](http://recyclerview.java:1232/))
	at androidx.recyclerview.widget.RecyclerView.setAdapter([RecyclerView.java:1194](http://recyclerview.java:1194/))
	at [mono.android](http://mono.android/).view.View_OnClickListenerImplementor.n_onClick(Native Method)
	at [mono.android](http://mono.android/).view.View_OnClickListenerImplementor.onClick([View_OnClickListenerImplementor.java:30](http://view_onclicklistenerimplementor.java:30/))
	at android.view.View.performClick([View.java:7862](http://view.java:7862/))
	at android.widget.TextView.performClick([TextView.java:15004](http://textview.java:15004/))
	at android.view.View.performClickInternal([View.java:7831](http://view.java:7831/))
	at android.view.View.access$3600([View.java:879](http://view.java:879/))
	at android.view.View$[PerformClick.run](http://performclick.run/)([View.java:29359](http://view.java:29359/))
	at android.os.Handler.handleCallback([Handler.java:883](http://handler.java:883/))
	at android.os.Handler.dispatchMessage([Handler.java:100](http://handler.java:100/))
	at android.os.Looper.loop([Looper.java:237](http://looper.java:237/))
	at [android.app](http://android.app/).ActivityThread.main([ActivityThread.java:8167](http://activitythread.java:8167/))
	at java.lang.reflect.Method.invoke(Native Method)
	at [com.android](http://com.android/).internal.os.RuntimeInit$[MethodAndArgsCaller.run](http://methodandargscaller.run/)([RuntimeInit.java:496](http://runtimeinit.java:496/))
	at [com.android](http://com.android/).internal.os.ZygoteInit.main([ZygoteInit.java:1100](http://zygoteinit.java:1100/))

**Comment by radimitrov** [comment]
I can't replicate this bug. What's your Android version?
As for the other stuff - not everyone will agree on the same design. For example I don't like reaching all the way up for common stuff like redo/undo.
I just try to provide a few customization options.

**Comment by Alisis33** [comment]
Android 10
One ui 2.5

I need a downgrade then, because I can not work with the present version anymore. The errors show up when ever I use selection or the new copy/paste

**Comment by radimitrov** [comment]
I'll upload a probable fix in ~30 minutes, should go live in a few hours.  Disabling the new "copy/paste in quickbar" option might be a quick fix/workaround on your end? But if that doesn't work then you might have to downgrade.

**Comment by Alisis33** [comment]
...I found the link where the bug come from. I use a non, standart sort keyboard. It calls "classic keyboard" if I use a other, the bug not show up. 

(The quickbar was allways is on and not made any problems before.)

**Comment by radimitrov** [comment]
Strange.. a keyboard shouldn't have any effect there. By the way I meant the option to just move the select toolbar back up top. That makes the quickbar not update often

**Comment by Alisis33** [comment]
Hm... this is a very strange bug... I changed the keyboard a cople timed, now it disappeared.
... I try if I can force it back. But maybe it was caused only by the keyboard(?)... I will write if I get more Infos.

**Comment by Alisis33** [comment]
...one issue remain, sometimes the copy-bar not show up if I select something. something must be writen first. but this bug was there before...

**Comment by radimitrov** [comment]
Yeah, I've figured out the the pattern to that one - just not how to fix it without breaking anything yet.
It happens when you close the selection toolbar when there is an active selection and then try to select again. One of the events doesn't fire. Going back to "cursor mode" resets it.

**Comment by Alisis33** [comment]
Yes, this I can reproduce...

The reported bug I thing was caused outside of csshell, after flip floping the keyboards, it never come back.
I found no fitting reports in internet.

But at least anyway a bug less ;-)

**Comment by Alisis33** [comment]
... the reported bug not show up again. This issue can be closed. If it come back, I will reopen it.

**Comment by radimitrov** [comment]
Well, something simple that should work as a fix is still in the new update. So it should be fine even when switching keyboards.
